indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -104,6 +94,8 @@ public final static InputRow parseInputRow(Writable value, InputRowParser parser
     if(parser instanceof StringInputRowParser && value instanceof Text) {
       //Note: This is to ensure backward compatibility with 0.7.0 and before
       return ((StringInputRowParser)parser).parse(value.toString());
+    } else if(value instanceof HadoopWritableInputRow) {
```

Comment:
It looks like we are adjusting things to make the Mapper understand more different types of objects, I think it might be cleaner to define the Mapper interface in terms of InputRow objects and move all of the parser and stuff to the InputFormat.

This would have the downside of making it a bit more difficult to take other InputFormats, but would have the plus of getting rid of the parser when the parser is not needed (it becomes a part of the InputFormat instead of the mapper).